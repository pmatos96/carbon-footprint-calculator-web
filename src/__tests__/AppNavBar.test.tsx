import AppNavBar from "@/app/appNavBar";
import { render } from "@testing-library/react";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
    useRouter: jest.fn(),
}));

describe("App Navbar Component Snapshot", () => {

  const mockPageItems = [{ name: 'HOME', route: '/' }, { name: 'CALCULATOR', route: '/calculator' }];
  const mockPush = jest.fn();
  (useRouter as jest.Mock).mockReturnValue({
    push: mockPush,
  });

  it("matches the snapshot", () => {

    const { asFragment } = render(
      <AppNavBar pageItems={mockPageItems} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
