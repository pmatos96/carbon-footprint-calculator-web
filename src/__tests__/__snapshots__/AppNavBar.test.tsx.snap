// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Navbar Component Snapshot matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-e0vmxu-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1tldm05-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1i3o8ml-MuiGrid2-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15a76qm-MuiSvgIcon-root"
          data-testid="SpaIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.55 12c-1.07-.71-2.25-1.27-3.53-1.61 1.28.34 2.46.9 3.53 1.61m10.43-1.61c-1.29.34-2.49.91-3.57 1.64 1.08-.73 2.28-1.3 3.57-1.64"
          />
          <path
            d="M15.49 9.63c-.18-2.79-1.31-5.51-3.43-7.63-2.14 2.14-3.32 4.86-3.55 7.63 1.28.68 2.46 1.56 3.49 2.63 1.03-1.06 2.21-1.94 3.49-2.63m-6.5 2.65c-.14-.1-.3-.19-.45-.29.15.11.31.19.45.29m6.42-.25c-.13.09-.27.16-.4.26.13-.1.27-.17.4-.26M12 15.45C9.85 12.17 6.18 10 2 10c0 5.32 3.36 9.82 8.03 11.49.63.23 1.29.4 1.97.51.68-.12 1.33-.29 1.97-.51C18.64 19.82 22 15.32 22 10c-4.18 0-7.85 2.17-10 5.45"
          />
        </svg>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
        >
          Carbon Footprint Calculator
        </h6>
      </div>
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTypography-root MuiTypography-button css-5e57lr-MuiTypography-root"
          >
            HOME
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1588512-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTypography-root MuiTypography-button css-5e57lr-MuiTypography-root"
          >
            CALCULATOR
          </span>
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
